tap/tcp_reader.go

```diff
@@ -101,6 +108,9 @@ func (reader *tcpReader) Read(p []byte) (int, error) {
 	}
 
 	l := copy(p, reader.data)
+	if !reader.isProtocolIdentified() {
```

Comment:
Stop hanging on to bytes when the protocol is already identified is a nice optimization.
Why not also drop `reader.buffer` when the protocol is identified?

Reply:
@nimrod-up9 it's actually done in here:

https://github.com/up9inc/mizu/blob/b0e95e2ea74ebdd26f85e880f0a42ebeeb28bb23/tap/tcp_stream.go#L72-L73